ui/pages/confirmations/hooks/useGasFeeInputs.js

```diff
@@ -118,6 +119,14 @@ export function useGasFeeInputs(
     ? retryTxMeta
     : _transaction;
 
+  const network = useSelector((state) =>
+    selectNetworkConfigurationByChainId(state, transaction?.chainId),
+  );
+
+  const networkClientId =
+    network?.rpcEndpoints?.[network?.defaultRpcEndpointIndex]
+      ?.networkClientId ?? transaction?.networkClientId;
```

Comment:
Can we remove the `transaction?.networkClientId` completely then? There is no possibility to having a transaction without `chainId` so this makes this branching not necessary.